### PR TITLE
Memberbox cleanup

### PR DESCRIPTION
**Pull request by rbri**
Another cleanup pr; have unified (and sync with the name of the delegate method) the naming of the methods in member box and added some more methods. Hope this will make the code more clean ans straightforward.

### CODE REVIEW

File: src/org/mozilla/javascript/FunctionObject.java [hunk 1]
```diff
@@ -91,7 +91,7 @@ public FunctionObject(String name, Member methodOrConstructor,
         }
         String methodName = member.getName();
         this.functionName = name;
-        Class<?>[] types = member.argTypes();
```
Comment (gbrail) [inline src/org/mozilla/javascript/FunctionObject.java]: Not on this line but...

If we're doing a "cleanup" PR, can we also clean up "import *" stuff? I see that this file and others have it. I'd love to update the checkstyle rule to disallow it (as we do in most Java codebases I've worked on) but there is too much of it in Rhino!

File: src/org/mozilla/javascript/NativeCallSite.java [hunk 2]
```diff
@@ -133,22 +133,7 @@ private Object js_toString(Scriptable obj)
         return sb.toString();
     }
 
-    private Object getUndefined()
```
Comment (gbrail) [inline src/org/mozilla/javascript/NativeCallSite.java]: Thanks! That old code is kind of like something from one of those "don't write code like this" web sites ;-)

**Comment by gbrail** [comment]
Here's another idea for ANOTHER PR... take a look at this:

https://github.com/mozilla/rhino/issues/462

Since you are already digging in to MemberBox.

The problem is that the user might try to instantiate, say, java.net.URLConnection and call a method on it. As far as the user is concerned they are only using an abstract interface. But the class that actually gets instantiated is a private class deep inside some sort of "sun" package, so the Java 9 module system logs a warning.

But once we put an object in MemberBox (or wherever) if we could introspect the method being called and walk the inheritance tree up to the "top" somehow, then we won't get that error. 

I don't know how else to explain that other than with a diagram but I think that's the idea!

File: src/org/mozilla/javascript/FunctionObject.java (hunk 1 above)
Comment (rbri) [reply src/org/mozilla/javascript/FunctionObject.java]: As you wish master ;-)

**Comment by rbri** [comment]
Sorry but i see no chance to work on this java9/10 stuff for the next weeks

**Comment by gbrail** [comment]
Thanks! I manually rebased and merged this.

**Pull request closed without merging**